nixos/modules/services/desktop-managers/cosmic.nix

```
@@ -0,0 +1,173 @@
+        default = true;
+      };
+    };
+
+    environment.cosmic.excludePackages = lib.mkOption {
+      description = "List of COSMIC packages to exclude from the default environment";
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      example = lib.literalExpression "[ pkgs.cosmic-player ]";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    # Environment packages
+    environment.pathsToLink = [
+      "/share/backgrounds"
+      "/share/cosmic"
+    ];
+    environment.systemPackages = utils.removePackagesByName (
+      with pkgs;
+      [
+        adwaita-icon-theme
+        alsa-utils
+        cosmic-applets
+        cosmic-applibrary
+        cosmic-bg
+        (cosmic-comp.override { useXWayland = cfg.xwayland.enable; })
+        cosmic-edit
+        cosmic-files
+        (config.services.displayManager.cosmic-greeter.package)
+        cosmic-icons
+        cosmic-launcher
+        cosmic-notifications
+        cosmic-osd
+        cosmic-panel
+        cosmic-player
+        cosmic-randr
+        cosmic-screenshot
+        cosmic-session
+        cosmic-settings
+        cosmic-settings-daemon
+        cosmic-term
+        cosmic-wallpapers
+        cosmic-workspaces-epoch
+        hicolor-icon-theme
+        playerctl
+        pop-icon-theme
+        pop-launcher
+        xdg-user-dirs
+      ]
+      ++ lib.optionals cfg.xwayland.enable [
+        xwayland
+      ]
+      ++ lib.optionals config.services.flatpak.enable [
+        cosmic-store
+      ]
+    ) config.environment.cosmic.excludePackages;
+
+    xdg = {
+      mime.enable = true;
+
+      icons = {
+        enable = true;
+        fallbackCursorThemes = lib.mkDefault [ "Cosmic" ];
+      };
+
+      portal = {
+        enable = true;
+        extraPortals = with pkgs; [
+          xdg-desktop-portal-cosmic
+          xdg-desktop-portal-gtk
+        ];
+        configPackages = lib.mkDefault ([
+          pkgs.xdg-desktop-portal-cosmic
+        ]);
+      };
+    };
+
+    systemd = {
+      packages = with pkgs; [ cosmic-session ];
+      user.targets = {
+        cosmic-session = {
+          wants = [ "xdg-desktop-autostart.target" ];
+          before = [ "xdg-desktop-autostart.target" ];
+        };
+        tray = {
+          description = "Cosmic Tray Target";
+          requires = [ "graphical-session-pre.target" ];
+        };
+      };
+    };
+
+    fonts.packages = utils.removePackagesByName (with pkgs; [
+      fira
+      noto-fonts
+      open-sans
+    ]) config.environment.cosmic.excludePackages;
+
+    # Required options for the COSMIC DE
+    environment.sessionVariables.X11_BASE_RULES_XML = "${config.services.xserver.xkb.dir}/rules/base.xml";
+    environment.sessionVariables.X11_EXTRA_RULES_XML = "${config.services.xserver.xkb.dir}/rules/base.extras.xml";
+    hardware.graphics.enable = true;
+    programs.dconf.packages = with pkgs; [ cosmic-session ];
+    security.pam.services.cosmic-greeter = { };
```

Please add descriptive comments explaining why each piece of code is present. For example, in the cosmic desktop module, we have:

```suggestion
    # Required for screen locker
    security.pam.services.cosmic-greeter = { };
```

Since there's a separate setting in the cosmic-greeter module for different reasons, these comments help clarify the distinct purposes and prevent accidental removal of necessary code.

---

Can you clarify a bit more on this? What options do you want me to add documentation for? Only `security.pam.services.cosmic-greeter` or any other options too?

---

Only for the options where the reasoning isn't obvious from an outside glance.

---

Do you mind listing the options you want documented?

---

I think this one only, for now. Also add it in cosmic-greeter module specifying that it is for authentication.